### PR TITLE
Make vc gen suite configurable

### PR DESCRIPTION
**Pull request by aljones15**
Removes all cryptosuites from data integrity pushing them out to their respective test projects
the only exception is eddsa-2022 which is maintained for backwards compatibility and should be removed
in a future major release.

Features:
1. changes the verifier test suite to take in a cryptosuite library or instance for test data generation
2. moves deps for cryptosuites to devDeps ensuring that test suites that use this lib don't end up with multiple versions of cryptosuites installed
3. adds VC 2.0 context to documentLoader
4. turns issuedVc and validVc into pure JSON

### CODE REVIEW

File: vc-generator/index.js [hunk 1]
```diff
@@ -20,35 +23,41 @@ const vcCache = new Map();
...
  * @param {string} options.suiteName - A suite name.
- * @param {string} options.keyType - A keyType.
+ * @param {object} options.key - A key.
+ * @param {object} options.cryptosuite - A cryptosuite for the test.
  * @param {Array<string>} options.mandatoryPointers - An array of JSON pointers.
  * @param {Array<string>} options.selectivePointers - An array of JSON pointers.
  * @param {boolean} options.verify - If a verify suite is needed.
  * @param {object} options.optionalTests - Options for running optional tests.
+ * @param {object} [options.testVector = validVc] - A credential to use for
+ * test data.
```
Comment (BigBlueHat) [inline vc-generator/index.js:33]:
```suggestion
 *   test data.
```
Indentation.

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
Just a wee whitespace tweak request at this point.

File: tests/fixtures/sdDoc.js
Comment (BigBlueHat) [inline tests/fixtures/sdDoc.js:1]: Can we store this as a `.json` file, since that's all it is?

File: index.js [hunk 2]
```diff
@@ -339,10 +339,14 @@ export function checkDataIntegrityProofVerifyErrors({
       }
       for(const endpoint of endpoints) {
         let name;
+        //FIXME remove this in MAJOR release and use testDataOptions.keyType
```
Comment (BigBlueHat) [inline index.js:342]:
```suggestion
        // FIXME remove this in MAJOR release and use testDataOptions.keyType
```

File: vc-generator/index.js [hunk 3]
```diff
@@ -1,8 +1,11 @@
 /*!
- * Copyright 2023 Digital Bazaar, Inc. All Rights Reserved
+ * Copyright 2023-24 Digital Bazaar, Inc. All Rights Reserved
```
Comment (BigBlueHat) [inline vc-generator/index.js:2]:
```suggestion
 * Copyright 2023 - 2024 Digital Bazaar, Inc. All Rights Reserved
```

Use whole years for date ranges.

File: vc-generator/secret.js [hunk 4]
```diff
@@ -1,33 +1,19 @@
...
-  return multikey.from(require(`${keyDir}/${keyPath}`));
-}
+const TEST_KEY_PAIR = {
+  id: 'did:key:z6MkwXG2WjeQnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT#z6MkwXG2Wje' +
+    'QnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT',
+  '@context': 'https://w3id.org/security/multikey/v1',
+  type: 'Multikey',
+  controller: 'did:key:z6MkwXG2WjeQnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT',
+  publicKeyMultibase: 'z6MkwXG2WjeQnNxSoynSGYU8V9j3QzP3JSqhdmkHc6SaVWoT',
+  secretKeyMultibase: 'zrv3rbPamVDGvrm7LkYPLWYJ35P9audujKKsWn3x29EUiGwwhdZ' +
```
Comment (BigBlueHat) [inline vc-generator/secret.js:15]: It'd be great to always leave a comment anywhere we store "secret" key material in public. Just something that explains that this is for testing only, don't use it, etc. Robots (and humans) will trip over this at some point, and when they do, we should be ready to explain why this is an exception to the "don't keep secrets in git" rule.

File: package.json [hunk 5]
```diff
@@ -6,7 +6,7 @@
   "type": "module",
   "scripts": {
     "lint": "eslint .",
-    "test": "mocha tests/"
+    "test": "mocha tests/ --timeout 25000"
```
Comment (BigBlueHat) [inline package.json:9]: Why the timeout increase?

File: package.json [hunk 6]
```diff
@@ -22,6 +22,10 @@
   },
   "homepage": "https://github.com/digitalbazaar/data-integrity-test-suite-assertion#readme",
   "devDependencies": {
+    "@digitalbazaar/bbs-2023-cryptosuite": "^1.2.0",
+    "@digitalbazaar/ecdsa-rdfc-2019-cryptosuite": "^1.0.1",
+    "@digitalbazaar/ecdsa-sd-2023-cryptosuite": "^3.2.1",
+    "@digitalbazaar/eddsa-rdfc-2022-cryptosuite": "^1.0.1",
```
Comment (BigBlueHat) [inline package.json:28]: Is there a reason these were moved to `devDependencies`?

**Review by BigBlueHat — CHANGES_REQUESTED** [review]
Some tweak requests and some questions.

File: package.json (hunk 5 above)
Comment (aljones15) [reply package.json:9]: bbs takes forever.

File: vc-generator/secret.js (hunk 4 above)
Comment (aljones15) [reply vc-generator/secret.js:15]: I improved the comment here: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/39/commits/09184ae4269bb6f113e3c94528bba8e6e8663f50

File: tests/fixtures/sdDoc.js
Comment (aljones15) [reply tests/fixtures/sdDoc.js:1]: I made it a parameter to the functionL https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/39/commits/09184ae4269bb6f113e3c94528bba8e6e8663f50

I am hoping to remove a default key entirely in a MAJOR release as this library really should just throw if you don't pass in a key to this suite.

File: package.json (hunk 6 above)
Comment (BigBlueHat) [reply package.json:28]: Then why are they still in the dependency list here?

File: tests/fixtures/sdDoc.js
Comment (BigBlueHat) [reply tests/fixtures/sdDoc.js:1]: I think this reply was to something else, maybe? I don't see `sdDoc.js` included anywhere (in this code at least), but would still like to see it be stored as JSON if possible.

**Review by BigBlueHat — CHANGES_REQUESTED** [review]

File: package.json (hunk 6 above)
Comment (aljones15) [reply package.json:28]: https://docs.npmjs.com/cli/v10/configuring-npm/package-json#devdependencies

> If someone is planning on downloading and using your module in their program, then they probably don't want or need to download and build the external test or documentation framework that you use.
> 
> In this case, it's best to map these additional items in a devDependencies object.
> 

The devDeps in this case are for the test project.

File: tests/fixtures/sdDoc.js
Comment (aljones15) [reply tests/fixtures/sdDoc.js:1]: sdDoc is now json: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/39/commits/ca1a065e0a007988be156d96da33869d006ac162

There is some pain involved with this. Even node 22's support of importing json is still experimental: https://nodejs.org/api/esm.html#json-modules

So I have to use `createRequire` or use node's `fs` to import json. Using the experimental feature `with {type: 'json'}` results in eslint breaking as eslint does not support features in the experimental stage. Weirdly, the parser they use esprima does not throw on `esprima.tokenize("import foo from './bar.json' with {type: 'json;}");`

Comment (BigBlueHat) [reply tests/fixtures/sdDoc.js:1]: Yeah, we do this already to import the other JSON in the test suites. My main goal is to make sure we use "the simplest thing that could possibly work"--i.e. don't use JavaScript when JSON is all that's needed.

Comment (aljones15) [reply tests/fixtures/sdDoc.js:1]: also of note, there is probably a small army of developers working on making es6 import statements work with json and other file types with out having to use `with` anyways, I moved the key, issuedVc, and validVc all to json here: https://github.com/w3c-ccg/data-integrity-test-suite-assertion/pull/39/commits/f772bf8807c26325a5fb66fe7cae00d8703dfc2c

I will need to check and make sure no test suite was somehow calling on validVc outside of this library (I just began exporting it in this PR so we should be safe). anyways, that should be it.

File: package.json (hunk 6 above)
Comment (BigBlueHat) [reply package.json:28]: For the tests that test this test project? 😜

File: index.js [hunk 7]
```diff
@@ -8,8 +8,13 @@ import {
   shouldBeBs58, verificationFail
 } from './helpers.js';
 import chai from 'chai';
+import {createRequire} from 'node:module';
 import {generateTestData} from './vc-generator/index.js';
-import {validVc} from './validVc.js';
+
+//FIXME remove this once node has non-experimental support
+//for es6 import and json
```
Comment (BigBlueHat) [inline index.js:15]:
```suggestion
```

That comment's going to be there for a very long time...and we use `createRequire()` throughout all these test suites. If someone cares to refactor it later, they can (and likely will). Just don't want to trip over a long lived todo list item.

File: tests/fixtures/keys/index.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
+/*!
+ * Copyright 2023-2024 Digital Bazaar, Inc.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import {createRequire} from 'node:module';
+
+//FIXME remove this once node has non-experimental support
+//for es6 import and json
```
Comment (BigBlueHat) [inline tests/fixtures/keys/index.js:8]:
```suggestion
```
Likewise.

File: vc-generator/secret.js [hunk 9]
```diff
@@ -1,33 +1,13 @@
 /*!
  * Copyright 2023-24 Digital Bazaar, Inc. All Rights Reserved
  */
+import * as Ed25519Multikey from '@digitalbazaar/ed25519-multikey';
 import {createRequire} from 'node:module';
-import {cryptosuites} from './constants.js';
 
+//FIXME remove this once node has non-experimental support
+//for es6 import and json
```
Comment (BigBlueHat) [inline vc-generator/secret.js:8]:
```suggestion
```

**Review by BigBlueHat — APPROVED** [review]
This looks good to go. I'd love those long lived `FIXME` comments taken out, but we can do that later/elsewhere if you need to get this merged sooner than later.

Thanks!

**Pull request merged**